### PR TITLE
incorporate the kind-projector plugin

### PR DESCRIPTION
**Pull request by jedws**
Almost all type-lambdas are rewritten using the kind-projector apart from the ones that it cannot support and those in the examples tree.

Also standardises a lot of formatting, in particular placing anonymous class impls on a new line.

### CODE REVIEW

**Comment by japgolly** [comment]
Nice work!

File: project/build.scala [hunk 1]
```diff
@@ -161,7 +161,8 @@ object build extends Build {
           }
         }
         </developers>
-      )
+      ),
+    addCompilerPlugin("org.spire-math" % "kind-projector_2.10" % "0.5.2")
```
Comment (japgolly) [inline project/build.scala]: Need Scala 2.11 plugin too right?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -242,13 +242,20 @@ but the consistency is worth it.
...
-    ({type λ[α]=F[X, α]})#λ
-    ({type λ[α, β]=F[X, α, β]})#λ
-    ({type l[a]=F[X, a]})#l
-    ({type l[a, b]=F[X, a, b]})#l
+    F[X, ?]     ===  ({type λ[α] = F[X, α]})#λ
+    F[X, ?, ?]  ===  ({type λ[α, β] = F[X, α, β]})#λ
+    F[X, ?]     ===  ({type l[a] = F[X, a]})#l
+    F[X, ?, ?]  ===  ({type l[a, b] = F[X, a, b]})#l
+
+When using Lambda Type Function syntax when kind-projector's inline syntax doesn't work, use greek letters for
```
Comment (japgolly) [inline CONTRIBUTING.md]: `When using Lambda Type Function syntax and the kind-projector's inline syntax doesn't work` clearer, no? Double-when without comma confused poor, fragile me.

Comment (jedws) [inline CONTRIBUTING.md]: yeah, hopefully fixed and added a link to the KP doco

File: project/build.scala (hunk 1 above)
Comment (jedws) [inline project/build.scala]: fixed

**Comment by aloiscochard** [comment]
Wow, awesome work @jedws !

**Comment by puffnfresh** [comment]
Build is failing. Think it's on the `check-gen-type-classes` sbt task.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -242,13 +242,22 @@ but the consistency is worth it.
...
+    F[X, ?]     ===  ({type λ[α] = F[X, α]})#λ
+    F[X, ?, ?]  ===  ({type λ[α, β] = F[X, α, β]})#λ
+    F[X, ?]     ===  ({type l[a] = F[X, a]})#l
+    F[X, ?, ?]  ===  ({type l[a, b] = F[X, a, b]})#l
+
+If using Lambda Type Function syntax (ie. when the kind-projector's inline syntax is insufficent), use greek letters for
+parameters.
+
+    λ[α => (α, α)]  ===  ({type λ[α] = (α, α)})#λ
+    λ[α[_] => F[α]  ===  ({type λ[α[_]] = F[α]})#λ
```
Comment (runarorama) [inline CONTRIBUTING.md]: Use `{{` brackets for proper code formatting in editors that are comment-aware?

Comment (jedws) [inline CONTRIBUTING.md]: @runarorama Not sure what you mean here, markdown is just four-space indent. I'm merely following the existing style…

**Comment by jedws** [comment]
thanks @puffnfresh think I've fixed it

**Pull request merged**